### PR TITLE
Update CONTRIBUTING.md with more workflow details.

### LINKED ISSUE #315: Define and document a better workflow.
In order to release, we should have a good, well-documented workflow for accepting contributions and for managing major/minor/micro releases.

I'm currently leaning towards the one-master flow described [here](http://endoflineblog.com/follow-up-to-gitflow-considered-harmful), but that's not at all set in stone, and I'd like to get some input before I document. I agree with the author's core claim that gitflow is over-engineered in a way that results in a lot of confusion and a history that's pretty hard to follow.

@grybmadsci, @fahhem, any thoughts?

### PR DESCRIPTION
**Pull request by jettisonjoe**
Fixes #315

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -46,14 +73,22 @@ making the review process as fast and painless as possible.
 OpenHTF's Python code follows the
 [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html).
 We provide a `pylintrc` file at the top level of our repo that you can use with
-[pylint](https://www.pylint.org/) to lint your Python code.
+[pylint](https://www.pylint.org/) to lint your Python code. We expect the 
+codebase to produce zero pylint warnings, although we allow the use of explicit
```
Comment (fahhem) [inline CONTRIBUTING.md]: ', and we allow' since `pylint: disable=...` comments don't create warnings

**Comment by fahhem** [comment]
Sweet, more docs!

**Comment by jettisonjoe** [comment]
Addressed feedback. PTAL.

**Comment by fahhem** [comment]
LGTM!

**Comment by jettisonjoe** [comment]
Thanks for the review!
